### ISSUE TITLE
Make it possible to register Blockly/Rules extensions through io-package.json instead of hardcoding here

### ISSUE THREAD

**Issue by AlCalzone** [ioBroker/ioBroker.javascript#807]
We already have something like this for the web adapter. It would be cool if 3rd party adapters could register blocks/rules the same way without requiring a JavaScript update each time.

**Comment by Apollon77** [comment]
Blockly this is already existing! see Telegram https://github.com/iobroker-community-adapters/ioBroker.telegram/blob/master/io-package.json#L271 and https://github.com/iobroker-community-adapters/ioBroker.telegram/blob/master/admin/blockly.js

For Rules I think the curent code logic is not allowing this ... so would be a feature request for Rules them or?!

**Comment by AlCalzone** [comment]
Yup, then feature request for rules. But maybe it could be consolidated somehow and standardized, so not every adapter dev has to learn how Blockly blocks and rules are written?

My dream solution would make this declarative, so an adapter could define a list of actions and declare for each
- Which inputs (if any) does the action have? Which type?
- Which outputs (if any) does the action have? Which type?
- Given a set of inputs, how does the generated JS code look like.

The hard parts (generating Blockly/Rules code) should be handled by the JS adapter.

**Comment by GermanBluefox** [comment]
I tried to implement it from the beginning, but without success. Because of that all blocks are now defined in javascript.

**Comment by AlCalzone** [comment]
Maybe we could revisit that decision or think about how we could solve it in the future? I'm just thinking about ways to make these features more accessible to developers.

I can only speak for myself but I'm probably not going to build blocks or rule actions given the amount of work and searching for Blockly docs that is currently necessary for it.
